packages/ffe-form-react/src/StepSelector.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import RadioButtonInputGroup from './RadioButtonInputGroup';
+import RadioButton from './RadioButton';
+import { arrayOf, string, shape, func, number } from 'prop-types';
+
+const StepSelector = ({
+    name,
+    options,
+    initialSelected,
+    label,
+    description,
+    onChange,
+}) => {
+    const initialValue = initialSelected ? Number(initialSelected.value) : null;
+    const [selectedValue, setSelectedValue] = useState(initialValue);
+
+    const optionSelected = value => {
+        const formattedValue = Number(value);
+        setSelectedValue(formattedValue);
+        onChange(formattedValue);
+    };
+
+    return (
+        <RadioButtonInputGroup
```

Comment:
Jeg tror ikke det er en god idé å abstrahere vekk `RadioButtonInputGroup` og `RadioButton` på denne måten.
Dette gjør nok løsningen mindre fleksibel fordi konsumenter mister kontrollen over disse komponentene.

Denne komponenten (altså StepSelector) har også en intern state som er både unødvendig og kompliserende.

JSX-koden i [forslaget til Peter](https://github.com/SpareBank1/designsystem/pull/761/files) ser mye mer "React-y" ut, og jeg tror det er en bedre vei å gå.
Ta gjerne utgangspunkt i den og tilpass styling og komponentnavn.

Reply:
Hvorfor jeg ikke legger opp til at folk bruker RadioButtonInputGroup selv ?

Fordi jeg må kontrollere radio knappene containeren for å style den.

Angående intern state, så valgte jeg det fordi at den originale komponenten brukte det og har blitt testet. Litt usikker på om checkbox+label vil funka i mobilbank osv. Men er ikke imot det, ville bare undersøke vil fungere. Virker som dekningen er grei, men som sagt så har den blitt testet masse med internal state så det er egentlig det tryggeste.

Reply:
Jeg syntes det er mye bedre om den er en helt selvstendig komponent som kan brukes som den er noe ala dette:

`<StepSelector {...props} />`

Istedenfor at man skal skrive alt selv hver gang. Da kunne man bare eksportert masse styling.

Reply:
> Fordi jeg må kontrollere radio knappene containeren for å style den.

Radioknappene har ikke noe direkte å gjøre med RadioButtonInputGroup, så det trenger du ikke nei. Nå ser jeg du har endret på koden, så du bare har laget custom radioknapper :+1: 

> Angående intern state, så valgte jeg det fordi at den originale komponenten brukte det og har blitt testet. Litt usikker på om checkbox+label vil funka i mobilbank osv. Men er ikke imot det, ville bare undersøke vil fungere. Virker som dekningen er grei, men som sagt så har den blitt testet masse med internal state så det er egentlig det tryggeste.

Checkbox+label? Kanskje det er litt tidlig på morningen for meg, men jeg hang ikke helt med på hvor det kom fra å snakke om checkbox+label.

Når det gjelder intern state så er det helt trygt, og jeg vil si kanskje tryggere, å ikke ha intern state. Siden den interne staten er borte så går jeg ikke mer inn på det nå, men vi kan godt ta en prat på chat hvis du vil diskutere det noe mer :) 

> Jeg syntes det er mye bedre om den er en helt selvstendig komponent som kan brukes som den er noe ala dette:
> `<StepSelector {...props} />`
> Istedenfor at man skal skrive alt selv hver gang. Da kunne man bare eksportert masse styling.

Når man lager en slik abstraksjon så lager du implementasjonsdetaljer og et api andre må forholde seg til. Dette passer sikkert for mange, men det er alltid noen som har sære behov. Da må de lage en prop av et slag for å tilpasse komponenten til seg, og apiet vokser. `ffe-tables-react` er et godt eksempel på på dette. Ta en titt på proptypes i `Table.js`. Jeg vil påstå at dette ikke er spesielt elegant.
Derfor mener jeg det er mye bedre at det vi lager i Designsystemet er enkelt, fleksibelt og composable.
Lokalt i appen din derimot må du gjerne lage en `StepSelector` og abstrahere så mye du vil til det som passer deres behov :)